### ISSUE TITLE
[Bug]: showItemInFolder hangs macOS Finder for about a minute

### ISSUE THREAD

**Issue by Nesvet** [electron/electron#38540]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

25.0.1

### What operating system are you using?

macOS

### Operating System Version

macOS Ventura 13.4

### What arch are you using?

x64

### Last Known Working Electron version

24.4.0

### Expected Behavior

```javascript
import { shell } from "electron";

shell.showItemInFolder("/path/to/file.ext");
```
This should open the folder and highlight the file in it

### Actual Behavior

In **renderer** process it leads to Finder's "Not Responding" state for about a minute, and then it show the file. Meanwhile in **main** process it works as expected

**Comment by jkleinsc** [comment]
@Nesvet As documented here: [https://www.electronjs.org/docs/latest/api/shell](https://www.electronjs.org/docs/latest/api/shell), shell is only available in a non-sandboxed renderer.  If you are seeing this issue in a non-sandboxed renderer please provide a gist testcase.

**Comment by codebytere** [comment]
Looks like this happened here: https://github.com/electron/electron/compare/v25.0.0-nightly.20230330...v25.0.0-nightly.20230331 - likely https://github.com/electron/electron/commit/3fa03977c7f5be9442b356ae39dd310276314dce

**Comment by rvion** [comment]
This feels like a pretty serious bug. Is there a workaround in the meantime ?

**Comment by mifi** [comment]
Maybe to disable nodeIntegration

**Comment by KevinHughes** [comment]
Here's my workaround, which runs AppleScript behind the scenes:
```
showItemInFolder(path) {
    let script = `
        set thePath to POSIX file "${path}" as string
        tell application "Finder" to reveal thePath
        tell application "Finder" to activate
    `;
    let process = require('child_process');
    let result = process.execSync("/usr/bin/osascript -", {
        input: script
    });
}
```

**Comment by KishanBagaria** [comment]
We define the function in main process now and invoke it from renderer. This can be done through https://www.npmjs.com/package/@electron/remote or [IPC](https://texts.blog/2022/04/20/simplifying-ipc-in-electron/).

**Comment by KevinHughes** [comment]
> We define the function in main process now and invoke it from renderer. This can be done through https://www.npmjs.com/package/@electron/remote or [IPC](https://texts.blog/2022/04/20/simplifying-ipc-in-electron/).

I see... I'l do this from now on. I've disabled remote myself for security reasons so am using IPC, but it's working great and would consider this issue closed.

The code for those wondering:

In main:
```
const { ipcMain, shell } = require('electron');
ipcMain.on('showItemInFolder', function(event, fullPath) {
    shell.showItemInFolder(fullPath);
});

```
In renderer:
```
const { ipcRenderer } = require('electron');
ipcRenderer.send('showItemInFolder', fullPath);
```

**Comment by bengotow** [comment]
Hey folks, just stumbled across this one as well -- I have a non-sandboxed renderer process and other methods in the `shell` module can be called successfully (eg: `require('electron').shell.openPath('...')`).   It seems this bug is specific to `showItemInFolder`, and hangs the Finder for 60s.  Until this is fixed, I think calling it from the main process only is the best bet.

Sidenote: I'm upgrading an Electron app from v24 => v30 and this is one of the only regressions I've found - impressive considering I'm behind 20+ months of updates!

**Comment by greggman** [comment]
If this function no longer works in the renderer and is not intending to be fixed, maybe it should just be removed?

**Comment by codebytere** [comment]
Hmm - is this still happening for folks? I ran this on macOS 15.4 Beta and it worked instantly.

**Comment by edo0xff** [comment]
Yes this is still happening. Calling `showItemInFolder` from render process hangs Finder, my solution: Use IPC Messages to expose it and call it from main process:

```jsx
// preload.js
const { contextBridge, ipcRenderer } = require('electron');

contextBridge.exposeInMainWorld('shell', {
  showItemInFolder: async (fullPath) => {
    try {
      await ipcRenderer.invoke('show-item-in-folder', fullPath);
    } catch (error) {
      console.error('Error showing item in folder:', error);
      throw error;
    }
  },
});
```
handle it in main process:

```jsx
import { app, BrowserWindow, ipcMain  } from 'electron';

// ...

const handleIpcShowItemInFolder = async (e, fullPath) => {
  shell.showItemInFolder(fullPath);
}

// ...

app.whenReady().then(() => {
  createWindow();

  ipcMain.handle('show-item-in-folder', handleIpcShowItemInFolder);

  app.on('activate', () => {
    if (BrowserWindow.getAllWindows().length === 0) {
      createWindow();
    }
  });
});
```
and call it in render.js or wherever you need it:

```jsx
// ...
window.shell.showItemInFolder(filePathWhitoutProtocol);
```
